modules/tensor_mechanics/include/materials/czm/CZMConstitutiveModelIncrementalBase.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CZMConstitutiveModelTotalBase.h"
+
+/**
+ *Base class used to implement traction separetion laws requiring an incremental formulation.
+ *For incremental cohesive constituve models, the user is responsble for calulating the interface
+ *traction increment and its derivatives w.r.t. the interface displacement jump. The interface
+ *traction increment and related derivatives should be implemented overriding the
+ *computeInterfaceTractionIncrementAndDerivatives method. The interface coordinate system assumes
+ *the three component of the traction and disaplcement jump being ordered as [N,S1,S2], where N is
+ *the normal component and S1, S2 two orthogonal tangential components. The model also assumes
+ *isotropic behavior in the tangential directions.
+ */
+
+class CZMConstitutiveModelIncrementalBase : public CZMConstitutiveModelTotalBase
```

Comment:
It is a bit wired to derive `Incremental` from `Total`. Can they derive from a same base class?